backend-ktor/src/main/kotlin/me/zwsmith/plugins/database/RecipeService.kt

```diff
@@ -85,39 +90,50 @@ class RecipeService(database: Database) {
     
     suspend fun updateRecipe(
         recipeId: Int,
-        name: String? = null,
-        description: String? = null,
-        ingredients: List<QuantifiedIngredient>? = null
-    ): Recipe = dbQuery {
-        var recipe = RecipeDao.findById(recipeId)
-            ?: throw IllegalArgumentException("Recipe with id $recipeId not found")
-        if (name != null) {
-            recipe.name = name
-        }
-        if (description != null) {
-            recipe.description = description
-        }
-        if(ingredients != null) {
-            updateRecipeIngredients(recipeId, ingredients)
-        }
-        recipe.toRecipe()
+        recipeUpdate: RecipeUpdate
+    ): Recipe? = dbQuery {
```

Comment:
I think its reasonable for this function to return an error result if the recipe id is not found